src/shared/utilities/timeoutUtils.ts

```diff
@@ -16,16 +18,18 @@ export class Timeout {
     private readonly timeoutLength: number
     private readonly timerPromise: Promise<void>
     private timerTimeout?: NodeJS.Timeout
-    private timerResolve?: (value?: void | PromiseLike<void> | undefined) => void
+    private timerResolve?: (value?: void | PromiseLike<undefined> | undefined) => void
+    private timerReject?: (value?: Error | PromiseLike<Error> | undefined) => void
```

Comment:
Does `Promise<T>` not work instead of `PromiseLike<T>`?

Reply:
It does, not sure why it was `PromiseLike` before. Updated the type definitions to better match the implementation.